Druid-Tests/DruidTestRTLCompiler.class.st

```diff
@@ -567,59 +561,45 @@ DruidTestRTLCompiler >> gen_primitiveAt [
 	jump10 := self Jump: 0.
 	currentBlock := self Label.
 	jump4 jmpTarget: currentBlock.
-	self MoveMw: 8 r: SPReg R: Extra3Reg.
-	self SubCq: 1 R: Extra3Reg.
-	self MoveR: Extra3Reg Mw: 8 r: SPReg.
+	self SubCq: 1 R: SendNumArgsReg.
 	self MoveCq: 8 R: ClassReg.
-	self MoveR: ReceiverResultReg R: SendNumArgsReg.
+	self MoveR: ReceiverResultReg R: Extra1Reg.
+	self AddR: Extra1Reg R: ClassReg.
+	self LogicalShiftLeftCq: 1 R: SendNumArgsReg.
 	self AddR: SendNumArgsReg R: ClassReg.
-	self MoveMw: 8 r: SPReg R: Extra3Reg.
-	self LogicalShiftLeftCq: 1 R: Extra3Reg.
-	self MoveR: Extra3Reg Mw: 8 r: SPReg.
-	self MoveMw: 8 r: SPReg R: Extra3Reg.
-	self AddR: Extra3Reg R: ClassReg.
-	self MoveM16: 0 r: ClassReg R: SendNumArgsReg.
-	self AndCq: 16rFFFF R: SendNumArgsReg.
-	self LogicalShiftLeftCq: 3 R: SendNumArgsReg.
-	self AddCq: 1 R: SendNumArgsReg.
-	self MoveR: SendNumArgsReg R: ReceiverResultReg.
+	self MoveM16: 0 r: ClassReg R: Extra1Reg.
```

Comment:
This looks like a pattern we could enhance with the canonicaliser

```smalltalk
self MoveCq: 8 R: ClassReg.
self MoveR: ReceiverResultReg R: Extra1Reg.
self AddR: Extra1Reg R: ClassReg.
self LogicalShiftLeftCq: 1 R: SendNumArgsReg.
self AddR: SendNumArgsReg R: ClassReg.
self MoveM16: 0 r: ClassReg R: Extra1Reg.
```

=>

```smalltalk
self MoveR: ReceiverResultReg R: ClassReg.
self LogicalShiftLeftCq: 1 R: SendNumArgsReg.
self AddR: SendNumArgsReg R: ClassReg.
self MoveM16: 8 r: ClassReg R: Extra1Reg.
```
or somehting like that. Probably interesting case for @hernanmd

Reply:
Even better, if we use a register as offset this could be further simplified.